crates/bevy_ecs/src/query/iter.rs

```diff
@@ -279,7 +279,7 @@ where
     /// It is always safe for shared access.
     unsafe fn fetch_next_aliased_unchecked<'a>(
         &mut self,
-    ) -> Option<[<Q::Fetch as Fetch<'a>>::Item; K]>
+    ) -> Option<[<Q::Fetch as Fetch<'a, 's>>::Item; K]>
```

Comment:
can we replace `a` lifetimes here with `w`?

Reply:
There is already an `'w` in scope from the self type. I think this is intentionally a different lifetime for this function.

Reply:
Ah yeah nice catch. In my `pipelined-rendering` impl I used the `w` lifetime directly instead of declaring a new one, which makes sense to me:

![image](https://user-images.githubusercontent.com/2694663/127227916-4a8686f9-b26a-440b-9af7-34f5a52261ec.png)

Reply:
that might be okay but one of the uses of `fetch_next_aliased_unchecked` does not want a borrow of `'w` they want a borrow tied to `&'_ self` so using `'a` seems more "correct" to me

Reply:
I'm not actually sure what the correct lifetime is here to be completely honest, using 'a here makes it seem like maybe we want a 'b for the state lifetime..?

Reply:
I think the usage of `'s` in this PR is correct for `QueryCombinationIter`, if we want to do `'a` -> `'w` it might be worth doing in a separate PR

Reply:
Yup based on our conversation here, the current impl is correct: https://discord.com/channels/691052431525675048/870067944640622612